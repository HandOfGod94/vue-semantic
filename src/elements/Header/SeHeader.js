import Vue from 'vue'
import Component from 'vue-class-component'

@Component({
  name: 'se-header',
  props: {
    as: { type: String, required: true },
    inverted: Boolean,
    icon: String,
    image: String,
    content: String,
    subheader: Boolean,
    block: Boolean,
    disabled: Boolean,
    align: String,
    justified: Boolean,
    color: String
  }
})
class SeHeader extends Vue {
  render (h) {
    let HeaderTag = this.as
    return (
      <HeaderTag class={this.classes}>
        {this.icon && <i class={`${this.icon} icon`}></i> }
        {this.image && <img src={this.image} class="ui image"/> }
        {this.content && <div class="content">{this.content} {this.$slots.subheader} </div>}
        {this.$slots.default}
        {!this.content && this.$slots.subheader}
      </HeaderTag>
    )
  }

  get classes () {
    let classes = {
      ui: true,
      inverted: this.inverted,
      [this.color]: this.color,
      [`${this.align} aligned`]: this.align,
      disabled: this.disabled,
      sub: this.subheader,
      block: this.block,
      justified: this.justified,
      header: true
    }
    return classes
  }
}

export default SeHeader
