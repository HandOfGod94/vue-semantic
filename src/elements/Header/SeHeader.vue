<template>
  <component :is="as" :class="classObject">
    <i v-if="icon" :class="`${icon} icon`"></i>
    <slot></slot>
    <div v-if="hasContent" class="content">
      <slot name="content">
        {{content}}
      </slot>
      <div v-if="hasSubheader" class="sub header" >
        <slot name="subheader">
          {{subheader}}
        </slot>
      </div>
   </div>
  </component>
</template>

<script>
export default {
  name: 'se-header',
  props: {
    as: { type: String, required: true },
    inverted: Boolean,
    sub: Boolean,
    block: Boolean,
    iconHeader: Boolean,
    icon: String,
    disabled: Boolean,
    align: String,
    justified: Boolean,
    color: String,
    content: String,
    subheader: String
  },
  computed: {
    classObject () {
      return {
        ui: true,
        inverted: this.inverted,
        [this.color]: this.color,
        [`${this.align} aligned`]: this.align,
        disabled: this.disabled,
        icon: this.iconHeader,
        sub: this.sub,
        block: this.block,
        justified: this.justified,
        header: true
      }
    },
    hasContent () { return !!this.$slots['content'] || this.content },
    hasSubheader () { return !!this.$slots['subheader'] || this.subheader }
  }
}
</script>

<style>

</style>
