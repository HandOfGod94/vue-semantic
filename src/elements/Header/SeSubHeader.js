import Vue from 'vue'
import Component from 'vue-class-component'

@Component({
  functional: true,
  name: 'se-sub-header'
})
class SeSubHeader extends Vue {
  render (h, context) {
    return (
      <div class="sub header">{context.children}</div>
    )
  }
}

export default SeSubHeader
