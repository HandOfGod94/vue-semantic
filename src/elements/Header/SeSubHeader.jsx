import Vue from 'vue'
import Component from 'vue-class-component'

@Component({
  name: 'se-sub-header'
})
class SeSubHeader extends Vue {
  render () {
    return (
      <div class="sub header">{this.$slots.default}</div>
    )
  }
}

export default SeSubHeader
