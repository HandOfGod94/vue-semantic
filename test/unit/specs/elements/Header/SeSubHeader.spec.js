import SeHeader from '@/elements/Header/SeHeader'
import SeSubHeader from '@/elements/Header/SeSubHeader'
import { mount } from '@vue/test-utils'

const SubHeaderContent = {
  render (h) {
    return (
      <SeSubHeader>This is subheader</SeSubHeader>
    )
  }
}

describe('SeSubHeader inside SeHeader', () => {
  it('should be inside SeHeader(class="sub header")', () => {
    const data = { propsData: { as: 'h1', subheader: true }, slots: { default: SubHeaderContent } }
    const wrapper = mount(SeHeader, data)
    expect(wrapper.contains('div.sub.header')).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
