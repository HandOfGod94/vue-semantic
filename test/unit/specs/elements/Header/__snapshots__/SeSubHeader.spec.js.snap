// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeSubHeader inside SeHeader should be inside SeHeader(class="sub header") 1`] = `
<h1 class="ui sub header">
    <div class="sub header">This is subheader</div>
</h1>
`;
